packages/beacon-node/src/chain/opPools/attestationPool.ts

```diff
@@ -178,14 +182,34 @@ export class AttestationPool {
 /**
  * Aggregate a new contribution into `aggregate` mutating it
  */
-function aggregateAttestationInto(aggregate: AggregateFast, attestation: phase0.Attestation): InsertOutcome {
+function aggregateAttestationInto(aggregate: AggregateFast, attestation: allForks.Attestation): InsertOutcome {
   const bitIndex = attestation.aggregationBits.getSingleTrueBit();
 
   // Should never happen, attestations are verified against this exact condition before
   if (bitIndex === null) {
     throw Error("Invalid attestation not exactly one bit set");
   }
 
+  if ("committeeBits" in attestation && !("committeeBits" in aggregate)) {
+    throw Error("Attempt to aggregate electra attestation into phase0 attestation");
+  }
+
+  if (!("committeeBits" in attestation) && "committeeBits" in aggregate) {
+    throw Error("Attempt to aggregate phase0 attestation into electra attestation");
+  }
+
+  if ("committeeBits" in attestation) {
+    // We assume attestation.committeeBits should already be validated in api and gossip handler and should be non-null
+    const attestationCommitteeIndex = attestation.committeeBits.getSingleTrueBit();
+    const aggregateCommitteeIndex = (aggregate as AggregateFastElectra).committeeBits.getSingleTrueBit();
+
+    if (attestationCommitteeIndex !== aggregateCommitteeIndex) {
```

Comment:
now same attestation data could belong to different committee indices so we need to add 1 more level to our cache
implemented in #6744